components/webui/imports/api/search/server/methods.js

```diff
@@ -25,37 +25,49 @@ const initSearchJobsDbManager = (sqlDbConnection, {searchJobsTableName}) => {
 };
 
 /**
- * Updates the search event when the specified job finishes.
+ * Modifies the search results metadata for a given job ID.
+ *
+ * @param {number} jobId
+ * @param {object} fields - The fields to be updated in the search results metadata.
+ */
+const updateSearchResultsMeta = (jobId, fields) => {
+    const filter = {
+        _id: jobId.toString(),
```

Comment:
To mitigate races, perhaps we should also take in and use the expected previous state in the filter?